src/plugins/FullScreenPlugin.js

```diff
@@ -0,0 +1,113 @@
+import { ButtonPlugin } from '../base-plugins/ButtonPlugin';
+import { Button } from '../ui-elements';
+
+/**
+ * A Springroll plugin to easily set up togglable fullscreen
+ */
+export class FullScreenPlugin extends ButtonPlugin {
+  
+  /**
+   *  Creates an instance of FullscreenPlugin
+   * 
+   * @param {string} targetElementSelector The selector for the element to be made fullscreen
+   * @param {string || HTMLElement } buttonSelector The selector for the button or the HTMLElement Button to be used to toggle fullscreen on the targetElement
+   */
+  constructor(buttonSelector) {
+    super({
+      name: 'fullScreen'
+    }); 
+
+    this._toggleButtons = [];
+    this.iFrame = null;
+
+    this.sendAllProperties = this.sendAllProperties.bind(this);
+
+    
+    this.toggleButton = document.querySelectorAll(buttonSelector);
+    this.toggleButton.forEach((button) => {
+      this._toggleButtons.push(new Button({
+        button: button,
+        onClick: this.toggleFullScreen.bind(this),
+        channel: 'fullScreen'
+      }));
+    });
+    
+
+    document.addEventListener('fullscreenchange',  () => {
+      this.sendProperty('fullScreen', this.isFullScreen);
```

Comment:
Is there a reason this one is different from the rest? `this.isFullScreen` returns an html element object, so the client just receives and object. You could just throw in the `this.sendAllProperties()` call instead.